File: bundles/org.eclipse.rap.rwt/src/org/eclipse/rap/rwt/widgets/TableColumn.java
/*******************************************************************************
 * Copyright (c) 2002-2006 Innoopract Informationssysteme GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Innoopract Informationssysteme GmbH - initial API and implementation
 ******************************************************************************/

package org.eclipse.rap.rwt.widgets;

import org.eclipse.rap.rwt.RWT;
import org.eclipse.rap.rwt.internal.widgets.ItemHolder;

public class TableColumn extends Item {

  private final Table parent;
  private int width = 0;

  public TableColumn( final Table parent, final int style ) {
    super( parent, checkStyle( style ) );
    this.parent = parent;
    ItemHolder.addItem( parent, this );
  }

  static int checkStyle( final int style ) {
    int result = RWT.NONE;
    if( style > 0 ) {
      result = style;
    }
    return result;
  }

  public Display getDisplay() {
    return parent.getDisplay();
  }

  public int getWidth() {
    return width;
  }

  public void setWidth( final int width ) {
    if( width >= 0 ) {
      this.width = width;
    }
  }
  
  
  ///////////////////////////////////
  // Methods to dispose of the widget

  protected void releaseChildren() {
  }

  protected void releaseParent() {
    TableItem[] items = parent.getItems();
    int index = parent.indexOf( this );
    for( int i = 0; i < items.length; i++ ) {
      items[ i ].removeText( index );
    }
    ItemHolder.removeItem( parent, this );
  }

  protected void releaseWidget() {
  }
}

File: bundles/org.eclipse.rap.rwt/js/org/eclipse/rap/rwt/widgets/TableColumn.js
/*******************************************************************************
 * Copyright (c) 2007 Innoopract Informationssysteme GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Innoopract Informationssysteme GmbH - initial API and implementation
 ******************************************************************************/

/**
 * This class provides the client-side counterpart for 
 * org.eclipse.rap.rwt.TableColumn.
 */
qx.OO.defineClass(
  "org.eclipse.rap.rwt.widgets.TableColumn",
  qx.ui.basic.Atom,
  function( parent ) {
    qx.ui.basic.Atom.call( this );
    this.setAppearance( "table-column" );
    this.setOverflow( qx.constant.Style.OVERFLOW_HIDDEN );
    // Add this column to the list of coluimns maintained by the table
    this._table = parent;
    this._table._addColumn( this );
    // Register mouse-listener add control the 'mouseover' appearance state
    this.addEventListener( "mouseover", this._onMouseOver, this );
    this.addEventListener( "mouseout",  this._onMouseOut, this );
  }
);

qx.Proto.dispose = function() {
  if ( this.getDisposed() ) {
    return true;
  }
  this.removeEventListener( "mouseover", this._onMouseOver, this );
  this.removeEventListener( "mouseout",  this._onMouseOut, this );
  return qx.ui.basic.Atom.prototype.dispose.call( this );
}

/**
 * This listener function is added and removed server-side
 */
qx.Proto.onClick = function( evt ) {
  org.eclipse.rap.rwt.EventUtil.widgetSelected( evt );
}

qx.Proto._onMouseOver = function( evt ) {
  this.addState( "mouseover" );
}

qx.Proto._onMouseOut = function( evt ) {
  this.removeState( "mouseover" );
}
